### ISSUE TITLE
It would be nice to have validation checks before starting deployment

### ISSUE THREAD

**Issue by arroyc** [Microsoft/azure-tools-for-java#97]
After 10 mins (approx) it throws error on missing arguments .. it should be good to do validation checks before deployment

![image](https://cloud.githubusercontent.com/assets/16231002/23080296/07a99a0e-f505-11e6-9ee6-31aca84ac7bb.png)

**Comment by milismsft** [comment]
The error is triggered when we find a Docker host but the credentials are missing or they are wrong (ie the corresponding Key Vault is missing or was deleted). It should be properly handled in the latest version.
